server/eclipse-project/src/main/java/us/freeandfair/corla/Main.java

```diff
@@ -80,24 +87,65 @@ public Main(final /*@ non_null */ Properties the_properties) {
   // Instance Methods
   
   /**
-   * Starts a ColoradoRLA server.
+   * Parse a port number from properties.
+   * 
+   * @param the_property The name of the property.
+   * @param the_default The default port number.
    */
-  public void start() {
-    LOGGER.info("starting server with properties: " + my_properties);
+  private int parsePortNumber(final String the_property, final int the_default) {
+    int result = the_default;
     
-    // get the port number from properties
-    int port_number = DEFAULT_PORT;
     try {
-      final int prop_port_number =
-          Integer.parseInt(my_properties.getProperty("port", String.valueOf(port_number)));
-      if (MIN_PORT <= prop_port_number && prop_port_number < MAX_PORT) {
-        port_number = prop_port_number;
+      final int prop_port =
+          Integer.parseInt(my_properties.getProperty(the_property, 
+                                                     String.valueOf(the_default)));
+      if (MIN_PORT <= prop_port && prop_port < MAX_PORT) {
+        result = prop_port;
       } else {
-        LOGGER.info("invalid port number in properties, using default port");
+        LOGGER.info("invalid port number in property " + the_property + 
+                    ", using default " + the_default);
       }
     } catch (final NumberFormatException e) {
-      LOGGER.info("could not read port number from properties, using default port");
+      LOGGER.info("could not read port number property " + the_property + 
+                  ", using default " + the_default);
+    }
+    
+    return result;
+  }
+  
+  /**
+   * Redirect a request from HTTP to HTTPS.
+   * 
+   * @param the_request The request.
+   * @param the_response The response.
+   * @param the_port The HTTPS port.
+   */
+  private void httpsRedirect(final Request the_request, final Response the_response,
```

Comment:
The body of this method is a fairly straightforward pure function `String -> String`, with some obvious possible edge cases. We could factor it out and add a few unit tests.

Alternately, we could use the Java `URL` class to parse the `String` we get from `request.url()`, then use the class again to construct a `URL` from exactly the parsed parts, modifying exactly the protocol and port, converting back to a `String` at the end. That way we can lean on the standard lib instead of implementing custom parsing.

I prefer the latter approach, since it will result in code that should be fairly obviously correct. What do you think?